docs/src/main/paradox/customization/generators.md

```diff
@@ -0,0 +1,68 @@
+# Generating pages with code
+
+You can make use of [source generators](https://www.scala-sbt.org/release/docs/Howto-Generating-Files.html) to define code that produces pages to be included in your paradox build.
+
+To do this you must define the `sourceGenerators` property for `paradoxMarkdownToHtml`. 
+
+The below example shows a `build.sbt` file that generates an extra markdown page via code:
+
+```scala
+lazy val root = (project in file(".")).
+  enablePlugins(ParadoxPlugin).
+  settings(
+    name := "Generator Project",
+    paradoxTheme := Some(builtinParadoxTheme("generic")),
+    sourceGenerators in (Compile, paradoxMarkdownToHtml) += Def.task {
+      val file = (sourceManaged in (Compile, paradox)).value / "generated-page.md"
+    
+      IO.write(file, "# Markdown generated by code!")
+    
+      Seq(file)
+    }.taskValue
+  )
+```
```

Comment:
I'd like to see these as tested code snippets (in scripted test, pulled into the doc) at some point, like we do in Cinnamon and elsewhere. It looks like we don't have this for other examples like this in the paradox docs, so fine to leave this as code directly for now. (And if we do that we can still have the generic theme in the snippet and then override sbt setting to no theme to simplify output in testing.)